libsolidity/codegen/YulUtilFunctions.cpp

```diff
@@ -2665,9 +2714,11 @@ string YulUtilFunctions::updateStorageValueFunction(
 		templ("dynamicOffset", !_offset.has_value());
 		templ("panic", panicFunction(PanicCode::Generic));
 		templ("value", suffixedVariableNameList("value_", 0, _fromType.sizeOnStack()));
-		if (_fromType.category() == Type::Category::Array)
+		if (_fromType.category() == Type::Category::Array || _fromType.category() == Type::Category::ArraySlice)
 			templ("copyToStorage", copyArrayToStorageFunction(
-				dynamic_cast<ArrayType const&>(_fromType),
+				_fromType.category() == Type::Category::Array ?
+				dynamic_cast<ArrayType const&>(_fromType) :
+				dynamic_cast<ArraySliceType const&>(_fromType).arrayType(),
```

Comment:
Why don't we pass the type itself?

Reply:
If it stays like this, can you add an assertion and comment that this only works properly for calldata array slices?

Reply:
Added